File: snapshots/output-net10.0/syntax/VBMain/Fields.vb
  Imports System.Diagnostics.CodeAnalysis
'         ^^^^^^ read_access scip-dotnet nuget . . System/
'                enclosing_range 0:8-0:26
'         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ import scip-dotnet nuget . . CodeAnalysis/
'                                         enclosing_range 0:8-0:39
'                ^^^^^^^^^^^ read_access scip-dotnet nuget . . Diagnostics/
'                            enclosing_range 0:8-0:26
'                            ^^^^^^^^^^^^ read_access scip-dotnet nuget . . CodeAnalysis/
'                                         enclosing_range 0:8-0:39

  Namespace VBMain
'           ^^^^^^ read_access scip-dotnet nuget . . VBMain/
'                  enclosing_range 2:10-2:16
      <SuppressMessage("ReSharper", "all")>
'      ^^^^^^^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 CodeAnalysis/SuppressMessageAttribute#`.ctor`().
'                      enclosing_range 3:5-3:20
      Public Class Fields
'                  ^^^^^^ definition scip-dotnet nuget . . VBMain/Fields#
'                         enclosing_range 3:4-4:23
'                         documentation ```vb\nClass Fields\n```
'                         kind Class
          Class Fields1
'               ^^^^^^^ definition scip-dotnet nuget . . VBMain/Fields#Fields1#
'                       enclosing_range 5:8-5:21
'                       documentation ```vb\nClass Fields1\n```
'                       kind Class
              Private ReadOnly Property1 As Integer
'                              ^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Fields#Fields1#Property1.
'                                        enclosing_range 6:12-6:49
'                                        documentation ```vb\nPrivate ReadOnly Fields1.Property1 As Integer\n```
'                                        kind Field
'                                        relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int32#
              Private Property2, Property3 As Int64
'                     ^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Fields#Fields1#Property2.
'                               enclosing_range 7:12-7:49
'                               documentation ```vb\nPrivate Fields1.Property2 As Long\n```
'                               kind Field
'                               relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int64#
'                                ^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Fields#Fields1#Property3.
'                                          enclosing_range 7:12-7:49
'                                          documentation ```vb\nPrivate Fields1.Property3 As Long\n```
'                                          kind Field
'                                          relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int64#
'                                             ^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 System/Int64#
'                                                   enclosing_range 7:44-7:49
              Private Property4 As Tuple(Of Char, Nullable(Of Integer))
'                     ^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Fields#Fields1#Property4.
'                               enclosing_range 8:12-8:69
'                               documentation ```vb\nPrivate Fields1.Property4 As Tuple(Of Char, Integer?)\n```
'                               kind Field
'                               relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Tuple#

              Public Sub New(ByVal field2 As Long, ByVal field3 As Long, ByVal field4 As Tuple(Of Char, Integer?), ByVal field1 As Integer)
'                        ^^^ definition scip-dotnet nuget . . VBMain/Fields#Fields1#`.ctor`().
'                            enclosing_range 10:12-10:137
'                            documentation ```vb\nPublic Sub Fields1.New(field2 As Long, field3 As Long, field4 As Tuple(Of Char, Integer?), field1 As Integer)\n```
'                            kind Constructor
'                                  ^^^^^^ definition scip-dotnet nuget . . VBMain/Fields#Fields1#`.ctor`().(field2)
'                                         enclosing_range 10:12-10:137
'                                         documentation ```vb\nfield2 As Long\n```
'                                         kind Parameter
'                                         relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int64#
'                                                        ^^^^^^ definition scip-dotnet nuget . . VBMain/Fields#Fields1#`.ctor`().(field3)
'                                                               enclosing_range 10:12-10:137
'                                                               documentation ```vb\nfield3 As Long\n```
'                                                               kind Parameter
'                                                               relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int64#
'                                                                              ^^^^^^ definition scip-dotnet nuget . . VBMain/Fields#Fields1#`.ctor`().(field4)
'                                                                                     enclosing_range 10:12-10:137
'                                                                                     documentation ```vb\nfield4 As Tuple(Of Char, Integer?)\n```
'                                                                                     kind Parameter
'                                                                                     relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Tuple#
'                                                                                                                        ^^^^^^ definition scip-dotnet nuget . . VBMain/Fields#Fields1#`.ctor`().(field1)
'                                                                                                                               enclosing_range 10:12-10:137
'                                                                                                                               documentation ```vb\nfield1 As Integer\n```
'                                                                                                                               kind Parameter
'                                                                                                                               relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int32#
                  Property2 = field2
'                 ^^^^^^^^^ write_access scip-dotnet nuget . . VBMain/Fields#Fields1#Property2.
'                           enclosing_range 11:16-11:25
'                             ^^^^^^ read_access scip-dotnet nuget . . VBMain/Fields#Fields1#`.ctor`().(field2)
'                                    enclosing_range 11:28-11:34
                  Property3 = field3
'                 ^^^^^^^^^ write_access scip-dotnet nuget . . VBMain/Fields#Fields1#Property3.
'                           enclosing_range 12:16-12:25
'                             ^^^^^^ read_access scip-dotnet nuget . . VBMain/Fields#Fields1#`.ctor`().(field3)
'                                    enclosing_range 12:28-12:34
                  Property4 = field4
'                 ^^^^^^^^^ write_access scip-dotnet nuget . . VBMain/Fields#Fields1#Property4.
'                           enclosing_range 13:16-13:25
'                             ^^^^^^ read_access scip-dotnet nuget . . VBMain/Fields#Fields1#`.ctor`().(field4)
'                                    enclosing_range 13:28-13:34
                  Property1 = field1
'                 ^^^^^^^^^ write_access scip-dotnet nuget . . VBMain/Fields#Fields1#Property1.
'                           enclosing_range 14:16-14:25
'                             ^^^^^^ read_access scip-dotnet nuget . . VBMain/Fields#Fields1#`.ctor`().(field1)
'                                    enclosing_range 14:28-14:34
              End Sub
          End Class
      End Class
  End Namespace

File: snapshots/output-net10.0/syntax/Main/Fields.cs
  using System.Diagnostics.CodeAnalysis;
//      ^^^^^^ read_access scip-dotnet nuget . . System/
//             enclosing_range 0:6-0:24
//      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ import scip-dotnet nuget . . CodeAnalysis/
//                                      enclosing_range 0:6-0:37
//             ^^^^^^^^^^^ read_access scip-dotnet nuget . . Diagnostics/
//                         enclosing_range 0:6-0:24
//                         ^^^^^^^^^^^^ read_access scip-dotnet nuget . . CodeAnalysis/
//                                      enclosing_range 0:6-0:37

  namespace Main;
//          ^^^^ read_access scip-dotnet nuget . . Main/
//               enclosing_range 2:10-2:14

  [SuppressMessage("ReSharper", "all")]
// ^^^^^^^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 CodeAnalysis/SuppressMessageAttribute#`.ctor`().
//                 enclosing_range 4:1-4:16
  public class Fields
//             ^^^^^^ definition scip-dotnet nuget . . Main/Fields#
//                    enclosing_range 4:0-31:1
//                    documentation ```cs\nclass Fields\n```
//                    kind Class
  {
      class Fields1
//          ^^^^^^^ definition scip-dotnet nuget . . Main/Fields#Fields1#
//                  enclosing_range 7:4-20:5
//                  documentation ```cs\nclass Fields1\n```
//                  kind Class
      {
          private readonly int Property1;
//                             ^^^^^^^^^ definition scip-dotnet nuget . . Main/Fields#Fields1#Property1.
//                                       enclosing_range 9:8-9:39
//                                       documentation ```cs\nprivate readonly int Fields1.Property1\n```
//                                       kind Field
//                                       relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int32#
          private Int64 Property2, Property3;
//                ^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 System/Int64#
//                      enclosing_range 10:16-10:21
//                      ^^^^^^^^^ definition scip-dotnet nuget . . Main/Fields#Fields1#Property2.
//                                enclosing_range 10:8-10:43
//                                documentation ```cs\nprivate long Fields1.Property2\n```
//                                kind Field
//                                relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int64#
//                                 ^^^^^^^^^ definition scip-dotnet nuget . . Main/Fields#Fields1#Property3.
//                                           enclosing_range 10:8-10:43
//                                           documentation ```cs\nprivate long Fields1.Property3\n```
//                                           kind Field
//                                           relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int64#
          private Tuple<char, Nullable<int>> Property4;
//                                           ^^^^^^^^^ definition scip-dotnet nuget . . Main/Fields#Fields1#Property4.
//                                                     enclosing_range 11:8-11:53
//                                                     documentation ```cs\nprivate Tuple<char, int?> Fields1.Property4\n```
//                                                     kind Field
//                                                     relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Tuple#

          public Fields1(long field2, long field3, Tuple<char, int?> field4, int field1)
//               ^^^^^^^ definition scip-dotnet nuget . . Main/Fields#Fields1#`.ctor`().
//                       enclosing_range 13:8-19:9
//                       documentation ```cs\npublic Fields1.Fields1(long field2, long field3, Tuple<char, int?> field4, int field1)\n```
//                       kind Constructor
//                            ^^^^^^ definition scip-dotnet nuget . . Main/Fields#Fields1#`.ctor`().(field2)
//                                   enclosing_range 13:8-19:9
//                                   documentation ```cs\nlong field2\n```
//                                   kind Parameter
//                                   relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int64#
//                                         ^^^^^^ definition scip-dotnet nuget . . Main/Fields#Fields1#`.ctor`().(field3)
//                                                enclosing_range 13:8-19:9
//                                                documentation ```cs\nlong field3\n```
//                                                kind Parameter
//                                                relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int64#
//                                                                   ^^^^^^ definition scip-dotnet nuget . . Main/Fields#Fields1#`.ctor`().(field4)
//                                                                          enclosing_range 13:8-19:9
//                                                                          documentation ```cs\nTuple<char, int?> field4\n```
//                                                                          kind Parameter
//                                                                          relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Tuple#
//                                                                               ^^^^^^ definition scip-dotnet nuget . . Main/Fields#Fields1#`.ctor`().(field1)
//                                                                                      enclosing_range 13:8-19:9
//                                                                                      documentation ```cs\nint field1\n```
//                                                                                      kind Parameter
//                                                                                      relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int32#
          {
              Property2 = field2;
//            ^^^^^^^^^ write_access scip-dotnet nuget . . Main/Fields#Fields1#Property2.
//                      enclosing_range 15:12-15:30
//                        ^^^^^^ read_access scip-dotnet nuget . . Main/Fields#Fields1#`.ctor`().(field2)
//                               enclosing_range 15:12-15:30
              Property3 = field3;
//            ^^^^^^^^^ write_access scip-dotnet nuget . . Main/Fields#Fields1#Property3.
//                      enclosing_range 16:12-16:30
//                        ^^^^^^ read_access scip-dotnet nuget . . Main/Fields#Fields1#`.ctor`().(field3)
//                               enclosing_range 16:12-16:30
              Property4 = field4;
//            ^^^^^^^^^ write_access scip-dotnet nuget . . Main/Fields#Fields1#Property4.
//                      enclosing_range 17:12-17:30
//                        ^^^^^^ read_access scip-dotnet nuget . . Main/Fields#Fields1#`.ctor`().(field4)
//                               enclosing_range 17:12-17:30
              Property1 = field1;
//            ^^^^^^^^^ write_access scip-dotnet nuget . . Main/Fields#Fields1#Property1.
//                      enclosing_range 18:12-18:30
//                        ^^^^^^ read_access scip-dotnet nuget . . Main/Fields#Fields1#`.ctor`().(field1)
//                               enclosing_range 18:12-18:30
          }
      }

      class Fields2
//          ^^^^^^^ definition scip-dotnet nuget . . Main/Fields#Fields2#
//                  enclosing_range 22:4-30:5
//                  documentation ```cs\nclass Fields2\n```
//                  kind Class
      {
          // Function pointer equivalent without calling convention
          unsafe delegate*<string, int> a;
//                                      ^ definition scip-dotnet nuget . . Main/Fields#Fields2#a.
//                                        enclosing_range 25:8-25:40
//                                        documentation ```cs\nprivate delegate*<string, int> Fields2.a\n```
//                                        kind Field
//                                        relationship type_definition local 0
          unsafe delegate*<delegate*<in string, int>, delegate*<ref string, ref readonly int>> b;
//                                                                                             ^ definition scip-dotnet nuget . . Main/Fields#Fields2#b.
//                                                                                               enclosing_range 26:8-26:95
//                                                                                               documentation ```cs\nprivate delegate*<delegate*<in string, int>, delegate*<ref string, ref readonly int>> Fields2.b\n```
//                                                                                               kind Field
//                                                                                               relationship type_definition local 1

          // Function pointer equivalent with calling convention
          unsafe delegate* managed<string, int> c;
//                                              ^ definition scip-dotnet nuget . . Main/Fields#Fields2#c.
//                                                enclosing_range 29:8-29:48
//                                                documentation ```cs\nprivate delegate*<string, int> Fields2.c\n```
//                                                kind Field
//                                                relationship type_definition local 0
      }
  }
